tests/test_datastore.py

```diff
@@ -447,6 +451,24 @@ def failNotImplemented(obj, path, ref):
                 else:
                     self.runIngestTest(failNotImplemented)
 
+    def testIngestSymlinkOfSymlink(self):
+        """Special test for symlink to a symlink ingest"""
+        mode = "symlink"
+        if mode not in self.ingestTransferModes:
+            return
+        metrics, ref = self._prepareIngestTest()
+        with lsst.utils.tests.getTempFilePath(".yaml") as realpath:
+            with open(realpath, 'w') as fd:
+                yaml.dump(metrics._asdict(), stream=fd)
+            with lsst.utils.tests.getTempFilePath(".yaml") as sympath:
+                os.symlink(os.path.abspath(realpath), sympath)
+                datastore = self.makeDatastore()
+                datastore.ingest(FileDataset(path=os.path.abspath(sympath), refs=ref), transfer=mode)
+                uri = ButlerURI(datastore.getUri(ref))
```

Comment:
Consider adding blank lines as follows:
```suggestion
                os.symlink(os.path.abspath(realpath), sympath)

                datastore = self.makeDatastore()
                datastore.ingest(FileDataset(path=os.path.abspath(sympath), refs=ref), transfer=mode)

                uri = ButlerURI(datastore.getUri(ref))
```
I think that would make it easier to follow the logic (setup, operation being tested, tests).